src/js/samples/components/Create/ReadSelector.tsx

```diff
@@ -0,0 +1,161 @@
+import { flatMap, includes, indexOf, toLower, without } from "lodash-es";
+import React, { useState } from "react";
+import { FetchNextPageOptions, InfiniteData, InfiniteQueryObserverResult } from "react-query/types/core/types";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import { getFontWeight, theme } from "../../../app/theme";
+import { Box, Icon, InputError, InputSearch, NoneFoundSection, Toolbar } from "../../../base";
+import { ScrollList } from "../../../base/ScrollList";
+import { StyledButton } from "../../../base/styled/StyledButton";
+import { FileResponse } from "../../../files/types";
+import ReadSelectorItem from "./ReadSelectorItem";
+
+type ReadSelectorBoxProps = {
+    error: string;
+};
+
+const ReadSelectorBox = styled(Box)<ReadSelectorBoxProps>`
+    ${props => (props.error ? `border-color: ${theme.color.red};` : "")};
+`;
+
+export const ReadSelectorButton = styled(StyledButton)`
+    min-width: 44px;
+`;
+
+const ReadSelectorError = styled(InputError)`
+    margin-bottom: 10px;
+`;
+
+const ReadSelectorHeader = styled.label`
+    align-items: center;
+    display: flex;
+    font-weight: ${getFontWeight("thick")};
+
+    label {
+        margin: 0;
+    }
+
+    span {
+        color: grey;
+        margin-left: auto;
+    }
+`;
+
+type ReadSelectorProps = {
+    data: InfiniteData<FileResponse>;
+    isFetchingNextPage: boolean;
+    fetchNextPage: (options?: FetchNextPageOptions) => Promise<InfiniteQueryObserverResult>;
+    isLoading: boolean;
+    onSelect: (selected: number[]) => void;
+    error: string;
+    selected: number[];
+};
+
+let selectedFiles = [];
```

Comment:
Mutable global variables are extremely dicey. Every instance of this function now shares the exact same list of selected files. Define it inside the function